### ISSUE TITLE
[GraphQL BUG] Data written to Dgraph don't show in GraphQL queries 

### ISSUE THREAD

**Issue by marvin-hansen** [dgraph-io/dgraph#4885]
### What version of Dgraph are you using?

Dgraph v2.0.0-rc1 Kubernetes installed with official Helm Charts 

### Have you tried reproducing the issue with the latest release?

No 

### What is the hardware spec (RAM, OS)?

6 Cores
24 GB memory 

### Steps to reproduce the issue (command/config used to run Dgraph).

1) Create GraphQL Schema with an m to n relation
2) Write some data to both types 
3) Query 

### Expected behaviour and actual result.

Expected:

When I write to the DB and the operation gets acknowledged, I expect to get my data back through a query. 

Actual result: 

When querying from any GraphQL Client, I get 
 
>     "message": "Non-nullable field 'plan_name' (type String!) was not present in result from Dgraph.  GraphQL error propagation triggered.",

From Ratel, I get the actual data. 

Query: 
> {
  node(func: uid(0xb6ea)) {
    uid
    expand(_all_) {
      uid
      expand(_all_)
    }
  }
}

Result: 
> {
  "data": {
    "node": [
      {
        "uid": "0xb6ea",
        "Peoplecovered.person": "Dewasa",
        "Peoplecovered.description": "Anda yang berusia 24 tahun ke atas",
        "Peoplecovered.plan": "Zurich EaZi Travel"
      }
    ]
  },

So, yes, the data are there, the fields are there, just the GraphQL endpoint isn't able to load them.

How do I debug this one?

What kind of logs / debug infos would help?

**Comment by marvin-hansen** [comment]
Minimal schema, mutations & query, just in case it helps. 


[dbg.zip](https://github.com/dgraph-io/dgraph/files/4291105/dbg.zip)

**Comment by pawanrawal** [comment]
Are you sure that the mutations are succeeding? I get the following error while running the `addPolicy` mutation.

```
{
  "errors": [
    {
      "message": "Cannot query field \"plan_policy\" on type \"Policy\".",
      "locations": [
        {
          "line": 5,
          "column": 7
        }
      ]
    },
    {
      "message": "Cannot query field \"plan_category\" on type \"Policy\". Did you mean \"plan_name\"?",
      "locations": [
        {
          "line": 6,
          "column": 7
        }
      ]
    }
  ]
}
```

which makes sense, since those fields are not there in the schema provided above for `Policy`.

**Comment by marvin-hansen** [comment]
Solved

I have rebuild the DB in question. Re the missing field, that was a copy & paste error of mine b/c the master schema is quite large with ~400 LoC so I only copied the affected types. 

The schema in question had a lot of daily changes, and as a result, I was bumping in this bug of changing types leading to duplicate definitions quite often and at some point, things stopped working. I think there is already an open issue in the tracker. 

I don't have enough time to investigate all these quibbles, therefore I scripted the 
DB rebuild and whenever DGraph starts glitching, it's gets a DROP ALL & rebuild.
